### PR TITLE
Pokedex branch 2 async

### PR DESCRIPTION
**Pull request by Sebs5384**
Replace .then() with async/await for better readability in our code.

### CODE REVIEW

File: cypress/e2e/pokedex.cy.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    cy.get("[data-cy='pokemon-search-box-list']").should("not.be.visible");
+
+    cy.get("[data-cy='pokemon-search-box']").click();
+    cy.get("[data-cy='pokemon-search-box-list']").should("be.visible");
+    cy.get("[data-cy='pokemon-search-box']").clear();
+    cy.get("[data-cy='pokemon-search-box']").type("Wartortle");
+
+    cy.get("[data-cy='Wartortle']").should("be.visible");
+  });
```
Comment (GonzaloAlt) [inline cypress/e2e/pokedex.cy.js]: Maybe you should isolate the tests a little bit more? One to search, another for closing the modal, etc.
Also, what is `1292`? can you save it in a variable? And could the title of the tests be more descriptive? `it('should do/show/etc something...', ()=>{})`

File: cypress/e2e/pokedex.cy.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  it("Clicks on pokemon card", () => {
```
Comment (GonzaloAlt) [inline cypress/e2e/pokedex.cy.js]: Can you put something more descriptive?

File: cypress/e2e/pokedex.cy.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    cy.get("[data-cy='weakness-icon']").should("have.attr", "src", "img/pokemon-types/icons/electric-type-icon.png");
+    cy.get("[data-cy='resistance-icon']").should("have.attr", "src", "img/pokemon-types/icons/fire-type-icon.png");
+    cy.get("[data-cy='retreat-icon']").children().should("have.length", 3);
+
+    cy.get("[data-cy='pokemon-description']").should("have.text", "A brutal POKéMON\nwith pressurized\nwater jets on its shell. They are\nused for high\nspeed tackles.");
+
+    cy.get("[data-cy='pokemon-modal-close-button']").click();
+  });
+
+  it("Uses paginator", () => {
```
Comment (GonzaloAlt) [inline cypress/e2e/pokedex.cy.js]: Same here.

File: cypress/e2e/pokedex.cy.js (hunk 1 above)
Comment (Sebs5384) [reply cypress/e2e/pokedex.cy.js]: I gave it a better description, isolated the test cases where the modal should respond to the close button and when the user uses the paginator search-box feature.

File: cypress/e2e/pokedex.cy.js (hunk 2 above)
Comment (Sebs5384) [reply cypress/e2e/pokedex.cy.js]: Added your feedback to my spec file, thank you !

File: cypress/e2e/pokedex.cy.js (hunk 3 above)
Comment (Sebs5384) [reply cypress/e2e/pokedex.cy.js]: Added it in here as well, thanks for the feedback !!

**Pull request merged**